### PR TITLE
[FIX] l10n_it_edi: EDI cron fix

### PR DESCRIPTION
**Pull request by lordkrandel**
The EDI cron was broken since we applied the batching to the EDI documents processing in this PR: #99227
Now we enable processing multiple invoices at a time.

### CODE REVIEW

File: addons/l10n_it_edi/models/account_edi_format.py [hunk 1]
```diff
@@ -1012,11 +1012,15 @@ def _l10n_it_post_invoices_step_2(self, invoices):
...
-        if not invoice.l10n_it_edi_transaction:
-            return self._l10n_it_post_invoices_step_1(invoice)
-        else:
-            return self._l10n_it_post_invoices_step_2(invoice)
+        # TODO: since batching is now possible, we should change
+        # the name of the invoice parameter to invoices
+        invoices1 = invoice.filtered(lambda x: not x.l10n_it_edi_transaction)
+        result1 = self._l10n_it_post_invoices_step_1(invoices1)
+
+        invoices2 = invoice.filtered(lambda x: x.l10n_it_edi_transaction)
```
Comment (jco-odoo) [inline addons/l10n_it_edi/models/account_edi_format.py:1020]: What if they got an l10n_it_edi_transaction in the meantime and what happens when in the return result then?

Comment (lordkrandel) [reply addons/l10n_it_edi/models/account_edi_format.py]: We discussed about it -  the batching avoids mixing the two groups, and we're changing the implementation to something mutually exclusive.

File: addons/l10n_it_edi/models/account_edi_format.py [hunk 2]
```diff
@@ -1023,8 +1023,9 @@ def _l10n_it_post_invoices_step_2(self, invoices):
         return to_return
 
     def _post_fattura_pa(self, invoice):
-        # OVERRIDE
-        if not invoice.l10n_it_edi_transaction:
+        if not invoice:
```
Comment (jco-odoo) [inline addons/l10n_it_edi/models/account_edi_format.py:1026]: @lordkrandel Is there a case where it can be empty?

Comment (lordkrandel) [reply addons/l10n_it_edi/models/account_edi_format.py:1026]: The method can be called from server actions and traceback

Comment (lordkrandel) [reply addons/l10n_it_edi/models/account_edi_format.py:1026]: Want me to remove it? Let's just merge this.

Comment (lordkrandel) [reply addons/l10n_it_edi/models/account_edi_format.py]: I don't really understand your way of thinking. 
Being defensive against unwanted parameters and conditions (especially when your method in Python is completely public and can be called from whatever code) is something I've been taught all the way from university to 10 years of profession.
I don't know where the confusion could come from: if you pass me no invoice, I'll post no invoice.
But whatever, I've removed it 🤷🏻

Comment (jco-odoo) [reply addons/l10n_it_edi/models/account_edi_format.py]: It depends: https://en.wikipedia.org/wiki/Design_by_contract

**Comment by jco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**